src/components/Table/table-cells/TableBodyCell.tsx

```diff
@@ -15,16 +14,67 @@ export const TableBodyCell: React.FC<TableCellProps> = ({
     className,
     iconStart,
     iconEnd,
-    isFirstColumn,
-    canExpand,
-    isExpanded,
-    depth,
-    getToggleRowExpandedProps,
+    expandableColumnIndex,
+    row,
+    cell,
     children,
     ...props
 }) => {
-    const paddingLeft = `${depth * 0.75}rem`;
-    const marginLeft = spacings.spacingMedium;
+    // Fix this type definition of any
+    const isExpandableColumn = (row as any)?.cells[expandableColumnIndex].value === cell?.value;
+    const leftMarginBuffer = 20; // px
+    const leftMarginIndentDepth = row?.depth * 20; // px
+
+    const renderIconTextIcon = () => (
+        <IconTextIcon
+            component="div"
+            iconStart={iconStart}
+            typographyProps={{isNowrap: true}}
+        >
+            {children}
+        </IconTextIcon>
+    );
+
+    const renderTableCell = () => {
+        // These are cells that are in the expandable row (canExpand) and it is the column in
+        // which the cells show the chevron icon to expand and collapse sub-rows (isExpandableColumn)
+        if (isExpandableColumn && row?.canExpand) {
+            return (
+                <TableCell
+                    {...
+                        (row?.canExpand && isExpandableColumn
+                            ? row?.getToggleRowExpandedProps({style: {marginLeft: `${leftMarginIndentDepth}px`}})
+                            : {}
+                        )
+                    }
+                >
+                    {row?.isExpanded
+                        ? <ChevronDown className="moonstone-marginRightNano"/>
+                        : <ChevronRight className="moonstone-marginRightNano"/>
+                    }
+                    {renderIconTextIcon()}
+                </TableCell>
+            );
+        }
+
+        // These are cells which are in the expandable column (isExpandableColumn), but themselves
+        // do now have sub-rows. Therefore, they need to have the appropriate nested indentation,
+        // but do not have the chevron to expand/collapse rows underneath them.
+        // Also, a buffer of 20px is added so that they are aligned with the cells that do have
+        // the chevron icons for expand/collapse
+        if (isExpandableColumn && !row?.canExpand) {
+            return (
+                <TableCell style={{marginLeft: `${leftMarginIndentDepth + leftMarginBuffer}px`}}>
+                    {renderIconTextIcon()}
+                </TableCell>
+            );
+        }
+
+        // These are just the normal cells in the other columns which don't display anything with
+        // relation to the row expansion feature
+        return <TableCell>{renderIconTextIcon()}</TableCell>;
```

Comment:
It's "funny" because in some cases we neither display icons nor a text in the cell, but a `Chip` for example. But I still don't find a better name

Reply:
Maybe changing the name of that render function to something like `renderContent()` would be less funny?

Reply:
I like the idea to change the name by something more generic.

Reply:
Changed the name of the function to `renderCellContent`